.github/workflows/netsim_runner.yaml

```diff
@@ -0,0 +1,254 @@
+        type: boolean
+        default: false
+      build_profile:
+        required: false
+        type: string
+        default: "release"
+      sim_paths:
+        required: false
+        type: string
+        default: "sims/iroh,sims/integration"
+      publish_metrics:
+        required: false
+        type: boolean
+        default: false
+      visualizations:
+        required: false
+        type: boolean
+        default: false
+      pr_number:
+        required: false
+        type: string
+        default: ""
+      report_table:
+        required: false
+        type: boolean
+        default: false
+  workflow_call:
+    inputs:
+      branch:
+        required: true
+        type: string
+      max_workers:
+        required: true
+        type: number
+        default: 4
+      netsim_branch:
+        required: true
+        type: string
+        default: "main"
+      debug_logs:
+        required: false
+        type: boolean
+        default: false
+      build_profile:
+        required: false
+        type: string
+        default: "release"
+      sim_paths:
+        required: false
+        type: string
+        default: "sims/iroh,sims/integration"
+      publish_metrics:
+        required: false
+        type: boolean
+        default: false
+      visualizations:
+        required: false
+        type: boolean
+        default: false
+      pr_number:
+        required: false
+        type: string
+        default: ""
+      report_table:
+        required: false
+        type: boolean
+        default: false
+
+env:
+  RUST_BACKTRACE: 1
+  RUSTFLAGS: -Dwarnings
+  MSRV: "1.66"
+  SCCACHE_GHA_ENABLED: "true"
+  RUSTC_WRAPPER: "sccache"
+  IROH_FORCE_STAGING_RELAYS: "1"
+
+jobs:
+  netsim:
+    name: Netsim
+    timeout-minutes: 45
+    runs-on: [self-hosted, linux, X64]
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+      with:
+        submodules: recursive
+        ref: ${{ inputs.branch }}
+
+    - name: Install rust stable
+      uses: dtolnay/rust-toolchain@stable
+
+    - name: Install sccache
+      uses: mozilla-actions/sccache-action@v0.0.5
+
+    - name: Build iroh
+      run: |
+        cargo build --profile ${{ inputs.build_profile }} --workspace --all-features
+
+    - name: Fetch and build chuck
+      run: |
+        cd ..
+        rm -rf chuck
+        git clone --single-branch --branch ${{ inputs.netsim_branch }} https://github.com/n0-computer/chuck.git
+        cd chuck
+        cargo build --release
+
+    - name: Install netsim deps
+      run: |
+        cd ../chuck/netsim
+        sudo apt update
+        ./setup.sh
+        ./cleanup.sh || true
+
+    - name: Copy binaries to right location
+      run: |
+        cp target/${{inputs.build_profile}}/iroh ../chuck/netsim/bins/iroh
+        cp target/${{inputs.build_profile}}/iroh-relay ../chuck/netsim/bins/iroh-relay
+        cp ../chuck/target/release/chuck ../chuck/netsim/bins/chuck
+
+    - name: Get commit sha
```

Comment:
kind of surprised github doesn't provide this already in some kind of variable

Reply:
Kinda does, but also kinda doesn't. Depending on how the job is called it uses a completely different set of `github.*.*` variable so it's pretty janky. This is one surefire way to get the commit hash you're currently checked out at.